python/dremio-flight/dremio/arguments/parse.py

```diff
@@ -13,116 +13,124 @@
-        type=bool,
-        help="Disables TLS server verification. Defaults to False.",
-        default=False,
-    )
-    parser.add_argument(
-        "-path_to_certs",
-        "--trustedCertificates",
-        dest="path_to_certs",
-        type=str,
-        help="Path to trusted certificates for encrypted connection. Defaults to system certificates.",
-        default=certifi.where(),
-    )
-    parser.add_argument(
-        "-sp",
-        "--sessionProperty",
-        dest="session_properties",
-        help="Key value pairs of SessionProperty, example: -sp schema='Samples.\"samples.dremio.com\"' -sp key=value",
-        required=False,
-        nargs="*",
-        action=KVParser,
-    )
-    parser.add_argument(
-        "-engine",
-        "--engine",
-        type=str,
-        help="The specific engine to run against. Only applicable to Dremio Cloud.",
-        required=False,
+        help="Path to config file. Defaults to config.yaml in the same directory.",
+        default="config.yaml",
     )
     return parser.parse_args()
+
+
+def convert_session_properties(session_properties: list[dict]) -> list[tuple]:
+    converted_properties = []
+    for prop in session_properties:
+        converted_properties += [
+            (k.encode("utf-8"), v.encode("utf-8")) for k, v in prop.items()
+        ]
+    return converted_properties
+
+
+def parse_config_file(config_rel_path: str):
+    with open(config_rel_path, mode="rt") as file:
+        try:
+            options = yaml.safe_load(file)
+            if "session_properties" in options:
+                options["session_properties"] = convert_session_properties(
+                    options["session_properties"]
+                )
+            return options
+        except yaml.YAMLError as exc:
+            raise exc
+
+
+def validate_options_type(config: dict):
+    for key in config:
+        value = config[key]
+        expected_type_for_key = options_default_validator["type"].get(key)
+
+        if expected_type_for_key is None:
+            continue
+
+        if not isinstance(value, expected_type_for_key):
+            raise TypeError(
+                f"The expected type for {key} is {expected_type_for_key}. You input {value} which is {type(value)}."
+            )
+
+
+def validate_required_options(config: dict):
+    for key in options_default_validator["required"]:
+        if config.get(key) is None:
+            raise Exception(
+                f"The '{key}' option is required. Please input a value for '{key}'."
+            )
+
+
+def validate_cloud_software_config(config: dict):
+    if config.get("username") is None:
+        if config.get("token") is None:
+            raise Exception(
+                "When connecting to Dremio Cloud, a 'token' must be supplied. If connecting to Dremio Software, a 'username' and ('password' or 'token') must be provided."
+            )
+    else:
+        if config.get("password") is None and config.get("token") is None:
```

Comment:
Use elif here instead of nesting if under the else block

Reply:
Whoops, should have collapsed that.